onnx_tf/frontend.py

```diff
@@ -28,6 +30,37 @@ class TensorflowFrontend(object):
   """ Tensorflow Frontend for ONNX
   """
 
+  @classmethod
+  def fold_constant(cls, onnx_graph):
+    """Fold constant nodes within an ONNX graph
+    TODO(tjingrant): do we need a special case to prevent folding identity?
+    """
+    for node in onnx_graph.nodes_proto:
+      # See if all inputs are present as contant tensors.
+      inclusion_mask = map(lambda x: x in onnx_graph.consts, node.input)
+      all_constant = all(inclusion_mask)
+      # If all inputs are constant, then fold this constant node.
+      if all_constant:
+        const_inputs = list(map(lambda x: onnx_graph.consts[x], node.input))
+        outputs = run_node(node, const_inputs)
+        # Make output tensors appear as graph initializers.
+        for index, output_name in enumerate(node.output):
+          output_content = outputs[index]
+          output_onnx_type = mapping.NP_TYPE_TO_TENSOR_TYPE[
```

Comment:
Use common tools to convert. (Or just pass dtype to `onnx_graph.add_..._proto_explicit` and convert it in that method.)